src/ui/public/courier/data_source/_abstract.js

```diff
@@ -1,14 +1,17 @@
 import _ from 'lodash';
 import angular from 'angular';
-import CourierRequestQueueProvider from 'ui/courier/_request_queue';
-import CourierErrorHandlersProvider from 'ui/courier/_error_handlers';
-import CourierFetchFetchProvider from 'ui/courier/fetch/fetch';
-import CourierDataSourceDecorateQueryProvider from 'ui/courier/data_source/_decorate_query';
+
+import 'ui/promises';
```

Comment:
I guess I added it because I noticed that this module requires/uses the `Promise` and `PromiseEmitter` services, but don't actually import them. It's been a while since I wrote this.